### PR TITLE
Update: reflect latest FIPS support status

### PR DESCRIPTION
**Pull request by fffonion**
### Description

 
From https://docs.google.com/document/d/1PJCcO_5DSTbv_08XxDlaS2Cx2-YeYfAencrFhS190AA/edit
and https://github.com/Kong/kong/pull/11725

### Testing instructions

Preview link: 

### Checklist 

- [ ] Review label added 
- [ ] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/gateway/kong-enterprise/fips-support/index.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ The following table explains where cryptographic algorithms are used for non-cry
 | core/uuid | Lua random number generator | Lua random number generator | The RNG isn’t used for cryptographic purposes. |
 | core/declarative_config/uuid | UUIDv5 (namespaced SHA1) | UUIDv5 (namespaced SHA1) | Used to generate a unique identifier. |
 | core/declarative_config/config_hash and core/hybrid/hashes | MD5 | MD5 | Used to generate a unique identifier. |
+{% if_version gte:3.5.x %}| core/request_id | rand(3) | rand(3) | The RNG isn’t used for cryptographic purposes. |{% endif_version %}
```
Comment (zhongweiy) [inline app/_src/gateway/kong-enterprise/fips-support/index.md:50]: shall we use "kong_request_id" or "request_id" here?

**Review by zhongweiy — APPROVED** [review]

**Comment by cloudjumpercat** [comment]
@fffonion I see the doc you linked mentions 3.4 FIPS status but the linked GitHub issue is for 3.5. I was just wondering if this change was implemented in 3.4 or 3.5. Thanks!

**Comment by fffonion** [comment]
@cloudjumpercat this PR include the change for the linked PR (which is targeted for 3.5), and some missing changes from previous versions (date back to 3.1).

File: app/_src/gateway/kong-enterprise/fips-support/index.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ content_type: reference
...
 
 
-{{site.ee_product_name}} provides a FIPS 140-2 compliant package for **Ubuntu 20.04**, **Ubuntu 22.04**, and **Red Hat Enterprise 8**. This package provides compliance for the core {{site.base_gateway}} product{% if_version gte:3.2.x %} and all out of the box plugins{% endif_version %}.
+{{site.ee_product_name}} provides a FIPS 140-2 compliant package for **Ubuntu 20.04** {% if_version gte:3.1.x %}, **Ubuntu 22.04** {% if_version gte:3.4.x %}, **Red Hat Enterprise 9** {% endif_version %}, and **Red Hat Enterprise 8** {% endif_version %}. This package provides compliance for the core {{site.base_gateway}} product {% if_version gte:3.2.x %} and all out of the box plugins {% endif_version %}.
 
-The package replaces OpenSSL, the primary SSL library in {{site.base_gateway}}, with [BoringSSL](https://boringssl.googlesource.com/boringssl/), which at its core uses the FIPS 140-2 validated BoringCrypto for cryptographic operations.
+The package uses the OpenSSL FIPS 3.0 module OpenSSL to provide FIPS 140-2 validated cryptographic operations.
```
Comment (cloudjumpercat) [inline app/_src/gateway/kong-enterprise/fips-support/index.md:12]: @fffonion Does this change correspond to a version? Just wondering if this should be conditionally rendered as well.

File: app/_src/gateway/kong-enterprise/fips-support/index.md [hunk 3]
```diff
@@ -18,8 +18,8 @@ The following table describes how {{site.base_gateway}} uses key derivation func
 
 | Component | Normal mode | FIPS mode | Notes |
 |-----------|-------------|-----------|-------|
-| core/rbac | bcrypt | PBKDF2 <sup>1</sup> | PBKDF2 in BoringSSL isn't FIPS validated. |
-| plugins/oauth2 <sup>2</sup> | Argon2 or bcrypt (when `hash_secret=true`) | Disabled (`hash_secret` can’t be set to `true`) | PBKDF2 in BoringSSL isn't FIPS validated. |
+| core/rbac | bcrypt | PBKDF2 <sup>1</sup> | Compliant via OpenSSL 3.0 FIPS provider  |
+| plugins/oauth2 <sup>2</sup> | Argon2 or bcrypt (when `hash_secret=true`) | Disabled (`hash_secret` can’t be set to `true`) | Compliant via OpenSSL 3.0 FIPS provider |
```
[the remark this answers is not in the document]
Comment (fffonion) [reply app/_src/gateway/kong-enterprise/fips-support/index.md:22]: The changes apply to all currently supported versions : )

File: app/_src/gateway/kong-enterprise/fips-support/index.md (hunk 2 above)
Comment (fffonion) [reply app/_src/gateway/kong-enterprise/fips-support/index.md:12]: Same as above, this applies to all versions

**Review by cloudjumpercat — APPROVED** [review]
LGTM! I just updated the conditional rendering formatting on one table. Thanks for making these changes!

**Pull request merged**